### PR TITLE
Improve nearsighted code

### PR DESCRIPTION
**Pull request by timothymtorres**
## About The Pull Request

This refactors the nearsighted code into it's own proc that was salvaged from #66306.  Just a code improvement with no game changes.

## Why It's Good For The Game

Improves the nearsighted code.  Nothing else was changed.

## Changelog

:cl: timothymtorres, TheBonded
code: Improve nearsighted code to be more robust
/:cl:

### CODE REVIEW

File: code/modules/mob/mob.dm [hunk 1]
```diff
@@ -1096,10 +1096,14 @@
 /mob/proc/has_nightvision()
 	return see_in_dark >= NIGHTVISION_FOV_RANGE
 
-///This mob is abile to read books
+/// This mob is abile to read books
```
Comment (Capsandi) [inline code/modules/mob/mob.dm:1099]:
```suggestion
/// This mob is able to read books
```

Comment (timothymtorres) [reply code/modules/mob/mob.dm]: Apparently I don't know how to read.

File: code/modules/mob/living/carbon/human/human.dm [hunk 2]
```diff
@@ -767,6 +767,16 @@
 	heat_exposure_stacks = 0
 	return ..()
 
+/mob/living/carbon/human/is_nearsighted()
+	if(!HAS_TRAIT(src, TRAIT_NEARSIGHT))
+		return FALSE
+
+	var/obj/item/clothing/glasses/eyewear = src.glasses
```
Comment (Mothblocks) [inline code/modules/mob/living/carbon/human/human.dm:774]:
```suggestion
	var/obj/item/clothing/glasses/eyewear = glasses
```

File: code/modules/mob/mob.dm [hunk 3]
```diff
@@ -1098,6 +1098,10 @@
 /mob/proc/has_nightvision()
 	return see_in_dark >= NIGHTVISION_FOV_RANGE
 
+/// Is this mob affected by nearsight
+/mob/proc/is_nearsighted()
+	return FALSE
```
Comment (MrMelbert) [inline code/modules/mob/mob.dm:1103]: The base level of `is_nearsighted()` should check for `HAS_TRAIT(src, TRAIT_NEARSIGHT)`. The nearsight trait isn't human exclusive (`become_nearsighted()` is a living level proc). 

Then in the human level proc just check for the paren't return value as well with the ol'
```
. = ..()
if(!.)
   return FALSE
```

**Review by MrMelbert — CHANGES_REQUESTED** [review]

File: code/modules/mob/mob.dm (hunk 3 above)
Comment (timothymtorres) [reply code/modules/mob/mob.dm]: Fixed with https://github.com/tgstation/tgstation/pull/66327/commits/58573b68f931a7678463ff2f797c86fb23c8f2d2 and  https://github.com/tgstation/tgstation/pull/66327/commits/67791dfff6ff35470e3793780e2ef172d968d037

File: code/modules/mob/living/carbon/human/human.dm [hunk 4]
```diff
@@ -767,6 +767,17 @@
...
+	. = ..() 
+	if(!.)
+		return FALSE
+
+	var/obj/item/clothing/glasses/eyewear = glasses
+	if(istype(eyewear) && eyewear.vision_correction)
+		return FALSE
+
+	return TRUE
+
```
Comment (LemonInTheDark) [inline code/modules/mob/living/carbon/human/human.dm:780]:
```suggestion
/mob/living/carbon/human/is_nearsighted()
	var/obj/item/clothing/glasses/eyewear = glasses
	if(istype(eyewear) && eyewear.vision_correction)
		return FALSE
	return ..()

```
Thoughts?

Comment (timothymtorres) [reply code/modules/mob/living/carbon/human/human.dm]: Looks robust and compact.  Very nice!

**Review by tralezab — APPROVED** [review]

**Pull request merged**